docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,191 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Extensions` are used by a particular group or several groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be installed and executed from `NEAR CLI`.
+- `Core NEAR CLI` extensions system only allows to introduce top-level commands through its design.
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-sub-account <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - with-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - with-generated-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-account <account-id> beneficiary <beneficiary-account-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - list-keys <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - add-key <account-id>
+    - use-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - generate-keypair network <"mainnet"|"testnet"|...>
```

Comment:
Same as my comment above where I'm not too sure the difference between `use-plaintext-public-key` and `generate-keypair`. Also for the `create-subaccount` section, we're using `with-plaintext-public-key` and `with-generated-keypair` but here we're using `use-plaintext-public-key` and just `generate-keypair` so perhaps we can keep it consistent?

Reply:
Is it any better now?

Reply:
I like this new naming much more. I'm fine with it as is but wanted to know your opinion on the possibility of saying `autogenerate-new-keypair` rather than `autogenerate-keypair`. It's provides a **bit** more context but I'm not sure it's worth it. What are your thoughts @frol

Reply:
Good point! I have also extended the command with two options: save the generated key to a keychain or just print it to the terminal.